code/modules/uplink/uplink_items/device_tools.dm

```diff
@@ -0,0 +1,227 @@
+			You will receive the unlock code upon activating the virus, and the new uplink may be charged with \
+			telecrystals normally."
+	item = /obj/item/cartridge/virus/frame
+	cost = 4
+	restricted = TRUE
+
+/datum/uplink_item/device_tools/failsafe
+	name = "Failsafe Uplink Code"
+	desc = "When entered the uplink will self-destruct immediately."
+	item = /obj/effect/gibspawner/generic
+	cost = 1
+	surplus = 0
+	restricted = TRUE
+	purchasable_from = ~(UPLINK_NUKE_OPS | UPLINK_CLOWN_OPS)
+
+/datum/uplink_item/device_tools/failsafe/spawn_item(spawn_path, mob/user, datum/uplink_handler/uplink_handler, atom/movable/source)
+	var/datum/component/uplink/uplink = source.GetComponent(/datum/component/uplink)
+	if(!uplink)
+		return
+	uplink.failsafe_code = uplink.generate_code()
+	var/code = "[islist(uplink.failsafe_code) ? english_list(uplink.failsafe_code) : uplink.failsafe_code]"
+	to_chat(user, span_warning("The new failsafe code for this uplink is now : [code]. You may check your antagonist info to recall this."))
+	return source //For log icon
+
+/datum/uplink_item/device_tools/toolbox
+	name = "Full Syndicate Toolbox"
+	desc = "The Syndicate toolbox is a suspicious black and red. It comes loaded with a full tool set including a \
+			multitool and combat gloves that are resistant to shocks and heat."
+	item = /obj/item/storage/toolbox/syndicate
+	cost = 1
+	illegal_tech = FALSE
+
+/datum/uplink_item/device_tools/rad_laser
+	name = "Radioactive Microlaser"
+	desc = "A radioactive microlaser disguised as a standard Nanotrasen health analyzer. When used, it emits a \
+			powerful burst of radiation, which, after a short delay, can incapacitate all but the most protected \
+			of humanoids. It has two settings: intensity, which controls the power of the radiation, \
+			and wavelength, which controls the delay before the effect kicks in."
+	item = /obj/item/healthanalyzer/rad_laser
+	cost = 3
+
+
+/datum/uplink_item/device_tools/suspiciousphone
+	name = "Protocol CRAB-17 Phone"
+	desc = "The Protocol CRAB-17 Phone, a phone borrowed from an unknown third party, it can be used to crash the space market, funneling the losses of the crew to your bank account.\
+	The crew can move their funds to a new banking site though, unless they HODL, in which case they deserve it."
+	item = /obj/item/suspiciousphone
+	restricted = TRUE
+	cost = 7
+	limited_stock = 1
+
+/datum/uplink_item/device_tools/binary
+	name = "Binary Translator Key"
+	desc = "A key that, when inserted into a radio headset, allows you to listen to and talk with silicon-based lifeforms, \
+			such as AI units and cyborgs, over their private binary channel. Caution should \
+			be taken while doing this, as unless they are allied with you, they are programmed to report such intrusions."
+	item = /obj/item/encryptionkey/binary
+	cost = 5
+	surplus = 75
+	restricted = TRUE
+
+// Low progression cost
+
+/datum/uplink_item/device_tools/emag
```

Comment:
Emag doesn't _really_ do much beyond niche features. I think at this point the emag functionality should just be parcelled with general 'hacking' stuff that traitors can do, or converting it into more objective outcomes/possibilities so that traitors aren't necessarily a good idea to just leave around to hack things.

Unless, of course, you're worried about emagged shuttles, which I can understand. But that might be another idea for an objective, no?